strawberryfields/backends/bosonicbackend/backend.py

```diff
@@ -277,39 +277,263 @@ def reset(self, pure=True, **kwargs):
+
+                z_max = int(
+                    np.ceil(
+                        np.sqrt(
+                            -4
+                            / np.pi
+                            * np.log(cutoff)
+                            * (1 + np.exp(-2 * epsilon))
+                            / (1 - np.exp(-2 * epsilon))
+                        )
+                    )
+                )
+                damping = 2 * np.exp(-epsilon) / (1 + np.exp(-2 * epsilon))
+
+                means_gen = it.tee(
+                    it.starmap(
+                        lambda l, m: l + 1j * m, it.product(range(-z_max, z_max + 1), repeat=2)
+                    ),
+                    2,
+                )
+                means = np.concatenate(
+                    (
+                        np.reshape(
+                            np.fromiter(means_gen[0], complex, count=(2 * z_max + 1) ** 2), (-1, 1)
+                        ).real,
+                        np.reshape(
+                            np.fromiter(means_gen[1], complex, count=(2 * z_max + 1) ** 2), (-1, 1)
+                        ).imag,
+                    ),
+                    axis=1,
+                )
+
+                weights = coef(means)
+                filt = weights > cutoff
+                weights = weights[filt]
+                means = means[filt]
+                weights /= np.sum(weights)
+                means *= 0.5 * damping * np.sqrt(np.pi * self.circuit.hbar)
+                cov = (
+                    0.5
+                    * self.circuit.hbar
+                    * (1 - np.exp(-2 * epsilon))
+                    / (1 + np.exp(-2 * epsilon))
+                    * np.identity(2)
+                )
+                cov = np.repeat(cov[None, :], weights.size, axis=0)
+
+                return weights, means, cov
+
+            elif desc == "complex":
+                raise ValueError("The complex description of GKP is not implemented")
+        else:
+            raise ValueError("Only square GKP are implemented for now")
 
     def prepare_fock(self, n, r=0.0001):
         """ Prepares the arrays of weights, means and covs of a Fock state"""
-        return
+        if r ** 2 > 1 / n:
+            raise ValueError("The parameter r**2={} is larger than n={}".format(r ** 2, n))
+        # A simple function to calculate the parity
+        parity = lambda n: 1 if n % 2 == 0 else -1
+        # All the means are zero
+        means = np.zeros([n + 1, 2])
+        covs = [
+            0.5
+            * self.circuit.hbar
+            * np.identity(2)
+            * (1 + (n - j) * r ** 2)
+            / (1 - (n - j) * r ** 2)
+            for j in range(n + 1)
+        ]
+        weights = np.array(
+            [
+                (1 - n * (r ** 2)) / (1 - (n - j) * (r ** 2)) * comb(n, j) * parity(j)
+                for j in range(n + 1)
+            ]
+        )
+        weights = weights / np.sum(weights)
+        return weights, means, covs
```

Comment:
I plotted the Wigner function for n=1 and did not see any negativity of the Wigner function.